support-modules/acquisition-events/src/main/scala/com/gu/support/acquisitions/ga/GoogleAnalyticsServiceImpl.scala

```diff
@@ -0,0 +1,255 @@
+import com.typesafe.scalalogging.LazyLogging
+import okhttp3.{Call, Callback, HttpUrl, OkHttpClient, Request, RequestBody, Response}
+
+import java.io.IOException
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.control.NonFatal
+import scala.util.matching.Regex
+
+object GoogleAnalyticsServiceImpl {
+  val clientIdPattern: Regex = raw"GA\d\.\d\.(\d+\.\d+)".r
+
+  // e.g. GUARDIAN_WEEKLY becomes GuardianWeekly
+  def camelCase(s: String): String = s
+    .split("_")
+    .map(_.toLowerCase.capitalize)
+    .mkString("")
+}
+
+trait GoogleAnalyticsService {
+  def submit(acquisition: AcquisitionDataRow, gaData: GAData, maxRetries: Int)(implicit ec: ExecutionContext): EitherT[Future, List[GAError], Unit]
+}
+
+object MockGoogleAnalyticsService extends GoogleAnalyticsService {
+  def submit(acquisition: AcquisitionDataRow, gaData: GAData, maxRetries: Int)(implicit ec: ExecutionContext): EitherT[Future, List[GAError], Unit] =
+    EitherT.fromEither(Right(()))
+}
+
+class GoogleAnalyticsServiceImpl(client: OkHttpClient) extends GoogleAnalyticsService with LazyLogging {
+  private[ga] val gaPropertyId: String = "UA-51507017-5"
+  private[ga] val endpoint: HttpUrl = HttpUrl.parse("https://www.google-analytics.com")
+
+  private[ga] def buildBody(acquisition: AcquisitionDataRow, gaData: GAData)(implicit ec: ExecutionContext): EitherT[Future, BuildError, RequestBody] = EitherT {
+    getAnnualisedValue(acquisition)
+      .fold(
+        error => {
+          logger.warn(s"Couldn't retrieve annualised value for this acquisition: $error")
+          buildPayload(acquisition, 0, gaData) // We still want to record the acquisition even if we can't get the AV
+        },
+        value => buildPayload(acquisition, value, gaData)
+      )
+      .map { maybePayload =>
+        maybePayload.map { payload =>
+          logger.debug(s"GA payload: $payload")
+          RequestBody.create(null, payload)
+        }
+      }
+  }
+
+  private[ga] def getAnnualisedValue(acquisition: AcquisitionDataRow)(implicit ec: ExecutionContext): EitherT[Future, String, Double] = {
+    acquisition.amount match {
+      case Some(amount) =>
+        val acquisitionModel = AcquisitionModel(amount.toDouble,
+          acquisition.product.value,
+          acquisition.currency.iso,
+          acquisition.paymentFrequency.value,
+          acquisition.paymentProvider.map(_.toString),
+          acquisition.printOptions.map(printOptions => PrintOptionsModel(printOptions.product.value, printOptions.deliveryCountry.alpha2))
+        )
+        EitherT(AnnualisedValueService.getAsyncAV(acquisitionModel, "ophan"))
+
+      case None => EitherT.fromEither[Future](Left("No amount for acquisition"))
+    }
+  }
+
+  private[ga] def getSuccessfulSubscriptionSignUpMetric(conversionCategory: ConversionCategory) =
+    conversionCategory match {
+      case _: ConversionCategory.ContributionConversion.type => ""
```

Comment:
since it's a case object, this should work
```suggestion
      case ConversionCategory.ContributionConversion => ""
```